### PR TITLE
i18n(import): roundtrip to get moved strings

### PR DESCRIPTION
**Pull request by exterkamp**
**Summary**
@connorjclark requested this roundtrip.

### CODE REVIEW

File: lighthouse-core/lib/i18n/locales/de.json [hunk 1]
```diff
@@ -1683,7 +1731,7 @@
     "message": "Einige Probleme haben diese Ausführung von Lighthouse beeinträchtigt:"
   },
   "lighthouse-core/report/html/renderer/util.js | varianceDisclaimer": {
-    "message": "Die Werte sind Schätzungen und können variieren. Die Leistungsbewertung [basiert nur auf diesen Messwerten](https://github.com/GoogleChrome/lighthouse/blob/d2ec9ffbb21de9ad1a0f86ed24575eda32c796f0/docs/scoring.md#how-are-the-scores-weighted)."
+    "message": "Values are estimated and may vary. The [performance score is calculated](https://web.dev/performance-scoring/) directly from these metrics."
```
Comment (brendankenny) [inline lighthouse-core/lib/i18n/locales/de.json:1734]: [\*ahem\*](https://github.com/GoogleChrome/lighthouse/pull/10754#discussion_r424109033)

Comment (connorjclark) [reply lighthouse-core/lib/i18n/locales/de.json:1734]: ah bummer

**Comment by patrickhulce** [comment]
> My assumption was the strings we moved would not require a re-translation

Yeah I even thought we confirmed this was a translation noop...

Now I will *really* never remember how tc works 😅

**Comment by brendankenny** [comment]
> > My assumption was the strings we moved would not require a re-translation
> 
> Yeah I even thought we confirmed this was a translation noop...
> 
> Now I will _really_ never remember how tc works 😅

No, you had it right, the strings that only moved are all translated. The ones newly in english were all changed, so tc defaults to english for those on our behalf until they're translated again.

If we wanted to keep e.g. the old translated `varianceDisclaimer` until translations for the new english string are available we'd have to either not do an import or manually add them back from the current locale files.

File: lighthouse-core/lib/i18n/locales/ar.json [hunk 2]
```diff
@@ -1310,17 +1340,29 @@
...
+  },
   "lighthouse-core/config/default-config.js | budgetsGroupDescription": {
     "message": "تضبط ميزانيات الأداء معايير لأداء موقعك الإلكتروني."
   },
   "lighthouse-core/config/default-config.js | budgetsGroupTitle": {
     "message": "الميزانيات"
   },
   "lighthouse-core/config/default-config.js | diagnosticsGroupDescription": {
-    "message": "مزيد من المعلومات حول أداء تطبيقك لا تؤثر هذه الأرقام [ بشكل مباشر في ](https://github.com/GoogleChrome/lighthouse/blob/d2ec9ffbb21de9ad1a0f86ed24575eda32c796f0/docs/scoring.md#how-are-the-scores-weighted) نتيجة الأداء."
+    "message": "مزيد من المعلومات حول أداء تطبيقك لا تؤثر هذه الأرقام [ بشكل مباشر في ](https://web.dev/performance-scoring/) نتيجة الأداء."
```
Comment (patrickhulce) [inline lighthouse-core/lib/i18n/locales/ar.json:1365]: wait, this is totally not gonna work as markdown link 😆

File: lighthouse-core/lib/i18n/locales/ar.json [hunk 3]
```diff
@@ -1332,7 +1374,7 @@
     "message": "تحسينات العرض الأول"
   },
   "lighthouse-core/config/default-config.js | loadOpportunitiesGroupDescription": {
-    "message": "يمكن أن تساعد هذه الاقتراحات على تحميل صفحتك بشكل أسرع. لا تؤثر هذه الاقتراحات [بشكل مباشر](https://github.com/GoogleChrome/lighthouse/blob/d2ec9ffbb21de9ad1a0f86ed24575eda32c796f0/docs/scoring.md#how-are-the-scores-weighted) في نتيجة الأداء."
+    "message": "يمكن أن تساعد هذه الاقتراحات على تحميل صفحتك بشكل أسرع. لا تؤثر هذه الاقتراحات [بشكل مباشر](https://web.dev/performance-scoring/) في نتيجة الأداء."
```
Comment (patrickhulce) [inline lighthouse-core/lib/i18n/locales/ar.json:1377]: or most of these arabic ones...

File: lighthouse-core/lib/i18n/locales/ar.json (hunk 2 above)
Comment (brendankenny) [reply lighthouse-core/lib/i18n/locales/ar.json:1365]: > wait, this is totally not gonna work as markdown link 😆

oh no, I wonder if something's wrong with our ctc -> lhl conversion...

Comment (exterkamp) [reply lighthouse-core/lib/i18n/locales/ar.json:1365]: That's just how github is with rtl. It works out in the end.
image

File: lighthouse-core/lib/i18n/locales/ar.json (hunk 3 above)
Comment (exterkamp) [reply lighthouse-core/lib/i18n/locales/ar.json:1377]: Answered above

File: lighthouse-core/lib/i18n/locales/ar.json (hunk 2 above)
Comment (brendankenny) [reply lighthouse-core/lib/i18n/locales/ar.json:1365]: > That's just how github is with rtl. It works out in the end.

ok, whew! Thanks @exterkamp

**Comment by exterkamp** [comment]
>No, you had it right, the strings that only moved are all translated. The ones newly in english were all changed, so tc defaults to english for those on our behalf until they're translated again.

^ this. There weren't many that purely moved. https://github.com/GoogleChrome/lighthouse/pull/10754#discussion_r424109033 is "correct" in that the strings from #10725 _would_ be a no-op, but they were never translated so they are still english. The strings that are new are english before and after this PR. The only regression is changed strings that haven't been translated yet, are now english. We can manually revert those. But that then makes us out of sync with tc/ etc. etc.

Options:
1. We can do nothing and wait 2 weeks and the changed strings and new strings will all be translated, then we can reimport.
2. We can manually pick out what strings we want to keep translated instead of import b/c the strings have been changed.
3. We can land this and some strings will regress.

**Comment by brendankenny** [comment]
> * We can manually pick out what strings we want to keep translated instead of import b/c the strings have been changed.

we could write a reverse `pruneObsoleteLhlMessages` to add back strings that are close enough they aren't worth getting rid of yet, but I think we'd only want to do that sometimes (when "close enough"), and that sounds tricky enough that it's probably worth waiting the two weeks instead :)

File: lighthouse-core/lib/i18n/locales/ar.json (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-core/lib/i18n/locales/ar.json:1365]: but it's like randomly in the middle of the string...OK github weird stuff haha

well phew!

**Comment by connorjclark** [comment]
jasmine updated these recently

**Pull request closed without merging**